sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -157,6 +161,11 @@ public class SystemSchema extends AbstractSchema
       .add("replication_factor", ColumnType.LONG)
       .build();
 
+  static final int SEGMENTS_SHARD_SPEC_INDEX = SEGMENTS_SIGNATURE.indexOf("shard_spec");
+  static final int SEGMENTS_DIMENSIONS_INDEX = SEGMENTS_SIGNATURE.indexOf("dimensions");
+  static final int SEGMENTS_METRICS_INDEX = SEGMENTS_SIGNATURE.indexOf("metrics");
+  static final int SEGMENTS_LAST_COMPACTION_STATE_INDEX = SEGMENTS_SIGNATURE.indexOf("last_compaction_state");
```

Comment:
nit: all of these can be `private static`

Reply:
I did a refactor that makes these unnecessary. The new logic moves `writeValueAsString` to `projectSegmentsRow`, and avoids both `ObjectMapper.writeValueAsString(o)` and `o.toString()` when `o` isn't being selected.